src/initializers/crawl-cleanup-permalinks.php

```diff
@@ -142,141 +164,56 @@ public function utm_redirect() {
-			global $post;
-			$proper_url = \get_permalink( $post->ID );
-
-			$page = \get_query_var( 'page' );
-			if ( $page && $page !== 1 ) {
-				$the_post   = \get_post( $post->ID );
-				$page_count = \substr_count( $the_post->post_content, '<!--nextpage-->' );
-				$proper_url = \user_trailingslashit( \trailingslashit( $proper_url ) . $page );
-				if ( $page > ( $page_count + 1 ) ) {
-					$proper_url = \user_trailingslashit( \trailingslashit( $proper_url ) . ( $page_count + 1 ) );
-				}
-			}
-
-			// Fix reply to comment links, whoever decided this should be a GET variable?
-			// phpcs:ignore WordPress.Security -- We know this is scary.
-			if ( isset( $_SERVER['REQUEST_URI'] ) && \preg_match( '`(\?replytocom=[^&]+)`', \sanitize_text_field( $_SERVER['REQUEST_URI'] ), $matches ) ) {
-				$proper_url .= \str_replace( '?replytocom=', '#comment-', $matches[0] );
-			}
-			unset( $matches );
-
-			// Prevent cleaning out posts & page previews for people capable of viewing them.
-			// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- We know this is scary.
-			if ( isset( $_GET['preview'] ) && isset( $_GET['preview_nonce'] ) && \current_user_can( 'edit_post' ) ) {
-				return;
-			}
-		}
-		elseif ( \is_front_page() ) {
-			if ( $this->current_page_helper->is_home_posts_page() ) {
-				$proper_url = \home_url( '/' );
-			}
-			elseif ( $this->current_page_helper->is_home_static_page() ) {
-				$proper_url = \get_permalink( $GLOBALS['post']->ID );
-			}
-		}
-		elseif ( $this->current_page_helper->is_posts_page() ) {
-			$proper_url = \get_permalink( \get_option( 'page_for_posts' ) );
-		}
-		elseif ( \is_category() || \is_tag() || \is_tax() ) {
-			$term = $wp_query->get_queried_object();
-			if ( \is_feed() ) {
-				$proper_url = \get_term_feed_link( $term->term_id, $term->taxonomy );
-			}
-			else {
-				$proper_url = \get_term_link( $term, $term->taxonomy );
-			}
-		}
-		elseif ( \is_search() ) {
-			$s          = \get_search_query();
-			$proper_url = \get_bloginfo( 'url' ) . '/?s=' . \rawurlencode( $s );
-		}
-		elseif ( \is_404() ) {
-			if ( \is_multisite() && ! \is_subdomain_install() && \is_main_site() ) {
-				if ( $current_url === \get_bloginfo( 'url' ) . '/blog/' || $current_url === \get_bloginfo( 'url' ) . '/blog' ) {
-					if ( $this->current_page_helper->is_home_static_page() ) {
-						$proper_url = \get_permalink( \get_option( 'page_for_posts' ) );
-					}
-					else {
-						$proper_url = \get_bloginfo( 'url' );
-					}
-				}
-			}
-		}
-		if ( ! empty( $proper_url ) && $wp_query->query_vars['paged'] !== 0 && $wp_query->post_count !== 0 ) {
-			if ( \is_search() ) {
-				$proper_url = \get_bloginfo( 'url' ) . '/page/' . $wp_query->query_vars['paged'] . '/?s=' . \rawurlencode( \get_search_query() );
-			}
-			else {
-				$proper_url = \user_trailingslashit( \trailingslashit( $proper_url ) . 'page/' . $wp_query->query_vars['paged'] );
-			}
+		$url_type = $this->crawl_cleanup_helper->get_url_type();
```

Comment:
I really like that you made this a switch.

Reply:
Thanks!